tests/ThrowingHttpClient.cs

```diff
@@ -0,0 +1,24 @@
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.HelixPoolProvider.Tests
+{
+    public class ThrowingHttpClient : HttpClient
+    {
+        private class ThrowingHttpMessageHandler : HttpMessageHandler
+        {
+            protected override Task<HttpResponseMessage> SendAsync(
+                HttpRequestMessage request,
+                CancellationToken cancellationToken)
+            {
+                throw new HttpRequestException("Test exception");
```

Comment:
We might want to make sure we handle the "timeout" style exceptions.

Reply:
I added a test for timeout handling